projects/observability/src/shared/components/cartesian/d3/data/range/cartesian-range.ts

```diff
@@ -0,0 +1,172 @@
+import { BaseType, select, Selection } from 'd3-selection';
+import { Area, area, curveMonotoneX, line, Line } from 'd3-shape';
+import { MouseDataLookupStrategy } from '../../../../utils/mouse-tracking/mouse-tracking';
+import { Range, Series } from '../../../chart';
+import { ChartTooltipTrackingOptions as ChartTooltipTrackingStrategy } from '../../../chart-interactivty';
+import { QuadtreeDataLookupStrategy } from '../../interactivity/data-strategy/quadtree-data-lookup-strategy';
+import { SingleAxisDataLookupStrategy } from '../../interactivity/data-strategy/single-axis-data-lookup-strategy';
+import { CartesianScaleBuilder } from '../../scale/cartesian-scale-builder';
+import { CartesianData } from '../cartesian-data';
+import { CartesianPoints } from '../series/cartesian-points';
+
+export class CartesianRange<TData> extends CartesianData<TData, Range<TData>> {
+  private static readonly CSS_RANGE_CLASS: string = 'range-group';
+  private static readonly CSS_SERIES_CLASS: string = 'range-data-series';
+  private static readonly LINE_WIDTH: number = 2;
+
+  public constructor(
+    protected readonly range: Range<TData>,
+    protected readonly scaleBuilder: CartesianScaleBuilder<TData>,
+    tooltipTrackingStrategy?: ChartTooltipTrackingStrategy
+  ) {
+    super(range, scaleBuilder, tooltipTrackingStrategy);
+  }
+
+  protected buildDataLookupStrategy(
+    visualization: Range<TData>,
+    strategy: ChartTooltipTrackingStrategy
+  ): MouseDataLookupStrategy<TData, Range<TData>> {
+    if (strategy.followSingleAxis !== undefined) {
+      return new SingleAxisDataLookupStrategy(
+        visualization,
+        this.getCombinedData(visualization),
+        this.xScale,
+        this.yScale,
+        strategy.radius,
+        strategy.followSingleAxis
+      );
+    }
+
+    return this.buildMultiAxisDataLookupStrategy();
+  }
+
+  private getCombinedData(range: Range<TData>): TData[] {
+    return [range.upper.data, range.lower.data].flatMap(d => d);
+  }
+
+  private getPairedData(range: Range<TData>): [TData, TData][] {
+    return range.upper.data.map((upperData: TData, index: number) => [upperData, range.lower.data[index]]);
+  }
+
+  /*
+   * SVG
+   */
+
+  public drawSvg(element: BaseType): void {
+    if (this.range.hide) {
+      return;
+    }
+
+    const group = select(element).append('g').classed(CartesianRange.CSS_RANGE_CLASS, true);
```

Comment:
Can we use our d3 selection wrapper here?

Reply:
Perhaps. I can try. Do you have an example?

Reply:
Check out `EntityNodeBoxRendererService` - it's basically just using our service's `select` method rather than the global function.